### PR TITLE
[Shop] Fix Forgot Password

### PR DESCRIPTION
**Pull request by inssein**
| Q               | A
| --------------- | ---
| Bug fix?        | yes
| New feature?    | no
| BC breaks?      | no
| Related tickets | fixes https://github.com/Sylius/Sylius/issues/6262
| License         | MIT

### CODE REVIEW

**Comment by tchapi** [comment]
Password reset still doesn't work because of https://github.com/Sylius/Sylius/issues/5859 : the route 'sylius_user_security_login' does not exist anymore and is called in UserController when resetting the password.

**Comment by inssein** [comment]
@tchapi not true in this case. You can override the route like done in this PR (https://github.com/Sylius/Sylius/pull/6357/files#diff-d64ce4611cdad270fd9f2da7a8749f3dR49).

File: src/Sylius/Bundle/CoreBundle/Resources/views/Email/passwordReset.html.twig [hunk 1]
```diff
@@ -3,5 +3,5 @@
 {% endblock %}
 
 {% block body %}
-    {{ 'sylius.email.password_reset.content'|trans({'%username%': user.username, '%url%': url('sylius_shop_request_password_reset_token', { 'token' : user.passwordResetToken})})|raw }}
+    {{ 'sylius.email.password_reset.content'|trans({'%username%': user.username, '%url%': url('sylius_shop_password_reset', { 'token' : user.passwordResetToken})})|raw }}
```
Comment (lchrusciel) [inline src/Sylius/Bundle/CoreBundle/Resources/views/Email/passwordReset.html.twig]: This file is placed in CoreBundle but it is coupled to shop-bundle. It should be moved out of here or a route should be declared both in core and shop bundle(with some other name)

**Comment by tchapi** [comment]
@inssein sure but still, a non-existent route in a Core file seems odd to me, no ? We shouldn't have to override anything to make this feature work, right ? (I mean, the global "reset password" feature)

**Comment by inssein** [comment]
@lchrusciel what do you suggest? I saw that problem, but it was already happening since all emails are in Core, for example: (https://github.com/inssein/Sylius/blob/fc4fc48c0a7776b1eee962bf34838bf213c2b13e/src/Sylius/Bundle/CoreBundle/Resources/views/Email/verification.html.twig)

@tchapi its not a non existent route in core, its a route in UserBundle that is supposed to be defined when integrated. I agree it seems silly, but thats how all the login stuff is currently done. I am open to modifying this PR so that it is done a better way.

**Comment by lchrusciel** [comment]
So all of this files are placed in a wrong bundle. CoreBundle should be usable without admin nor shop bundle. But it is not with the current implementation.

**Comment by inssein** [comment]
@lchrusciel having a hard time thinking of where to move them. I initially thought ShopBundle, but then API related actions should also send these emails. CoreBundle is the only thing connecting these bundles.

Maybe there is a way to optionally couple these bundles?

**Comment by michalmarcinkowski** [comment]
Thank you Hussein! 👍 

We will try to decouple the routing in a separate PR.

**Pull request merged**